packages/xchain-ethereum/src/client.ts

```diff
@@ -325,36 +325,47 @@ export default class Client implements XChainClient, EthereumClient {
       // Follow approach is only for testnet
       // For mainnet, we will use ethplorer api(one request only)
       // https://github.com/xchainjs/xchainjs-lib/issues/252
-      return Promise.all(
-        newAssets.map(async (asset) => {
-          if (assetToString(asset) !== assetToString(AssetETH)) {
-            // Handle token balances
-            const assetAddress = getTokenAddress(asset)
-            if (!assetAddress) {
-              throw new Error('Invalid asset')
-            }
-
-            const balance = await etherscanAPI.getTokenBalance({
-              baseUrl: this.etherscan.baseUrl,
-              address: ethAddress,
-              assetAddress,
-              apiKey: this.etherscan.apiKey,
-            })
-            const decimals = await this.call<BigNumberish>(assetAddress, erc20ABI, 'decimals', [])
-            return {
-              asset,
-              amount: baseAmount(balance.toString(), BigNumber.from(decimals).toNumber() || ETH_DECIMAL),
-            }
-          } else {
-            // Handle ETH balances
-            const balance = await this.etherscan.getBalance(ethAddress)
-            return {
-              asset: AssetETH,
-              amount: baseAmount(balance.toString(), ETH_DECIMAL),
-            }
+      // And to avoid etherscan api call limit, it gets balances in a sequence way, not in parallel
+      const balances = []
+      for (let i = 0; i < newAssets.length; i++) {
+        const asset = newAssets[i]
+        if (assetToString(asset) !== assetToString(AssetETH)) {
+          // Handle token balances
+          const assetAddress = getTokenAddress(asset)
+          if (!assetAddress) {
+            throw new Error('Invalid asset')
           }
-        }),
-      )
+
+          const balance = await etherscanAPI.getTokenBalance({
+            baseUrl: this.etherscan.baseUrl,
+            address: ethAddress,
+            assetAddress,
+            apiKey: this.etherscan.apiKey,
+          })
+          const decimals = await this.call<BigNumberish>(assetAddress, erc20ABI, 'decimals', [])
+          balances.push({
+            asset,
+            amount: baseAmount(balance.toString(), BigNumber.from(decimals).toNumber() || ETH_DECIMAL),
+          })
+        } else {
+          // Handle ETH balances
+          const balance = await this.etherscan.getBalance(ethAddress)
+          balances.push({
+            asset: AssetETH,
+            amount: baseAmount(balance.toString(), ETH_DECIMAL),
+          })
+        }
+        // Due to etherscan api call limitation, put some delay before another call
+        // Free Etherscan api key limit: 5 calls per second
+        // So 0.3s delay is reasonable for now
+        await new Promise((resolve) => {
```

Comment:
^ You can use [`delay` helper from xchain-util ](https://github.com/xchainjs/xchainjs-lib/blob/03ba565f6e87315aab2ec0f3e19bd48e3bec8b68/packages/xchain-util/src/async.ts#L19)